test-data/unit/check-unreachable-code.test

```diff
@@ -457,3 +458,44 @@ else:
 reveal_type(x)  # E: Revealed type is 'builtins.str'
 [builtins fixtures/ops.pyi]
 [out]
+
+[case testShortCircuitInExpression]
+import typing
+def f() -> bool: pass
+PY2 = f()
+PY3 = f()
+
+a = PY2 and 's'
```

Comment:
@ilevkivskyi That doesn't work, though, since `infer_condition_value` doesn't take constants into account...